File: 16-3sum-closest/3sum-closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
     nums.sort((a,b)=> a-b);
    let resultSum = nums[0]+nums[1]+nums[2]
    let minDiff = Number.MAX_SAFE_INTEGER
    
    for(let i = 0; i< nums.length-2;i++) {
        let l = i+1;
        let r = nums.length - 1;
        
        while(l < r){
            let sum = nums[i] + nums[l] + nums[r];
          if(sum == target){
              return sum;
          } 
        else if(sum < target){
            l++;
        }
        else{
            r--;
        }
        let diffTar = Math.abs(sum-target);
        if(diffTar < minDiff)
            {
                resultSum = sum;
                minDiff = diffTar;
            }
    }
}
       return resultSum
}
        

File: 16-3sum-closest/3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        closetSum = sum(nums[:3])
        for i in range(len(nums)-2):
            l,r = i+1, len(nums)-1
            while l < r:
                sums = nums[i] + nums[l] + nums[r]
                if abs(target - sums) < abs(target - closetSum):
                    closetSum = sums

                if sums == target:
                    return target
                elif sums < target:
                    l += 1
                else:
                    r -= 1
        return closetSum

                
                
        